compiler/rustc_middle/src/ty/vtable.rs

```diff
@@ -1,17 +1,32 @@
 use std::convert::TryFrom;
+use std::fmt;
 
 use crate::mir::interpret::{alloc_range, AllocId, Allocation, Pointer, Scalar, ScalarMaybeUninit};
-use crate::ty::fold::TypeFoldable;
-use crate::ty::{self, DefId, SubstsRef, Ty, TyCtxt};
+use crate::ty::{self, Instance, PolyTraitRef, Ty, TyCtxt};
 use rustc_ast::Mutability;
 
-#[derive(Clone, Copy, Debug, PartialEq, HashStable)]
+#[derive(Clone, Copy, PartialEq, HashStable)]
 pub enum VtblEntry<'tcx> {
     MetadataDropInPlace,
     MetadataSize,
     MetadataAlign,
     Vacant,
-    Method(DefId, SubstsRef<'tcx>),
+    Method(Instance<'tcx>),
+    TraitVPtr(PolyTraitRef<'tcx>),
+}
+
+impl<'tcx> fmt::Debug for VtblEntry<'tcx> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            VtblEntry::MetadataDropInPlace => write!(f, "MetadataDropInPlace")?,
+            VtblEntry::MetadataSize => write!(f, "MetadataSize")?,
+            VtblEntry::MetadataAlign => write!(f, "MetadataAlign")?,
+            VtblEntry::Vacant => write!(f, "Vacant")?,
+            VtblEntry::Method(instance) => write!(f, "Method: {}", instance)?,
+            VtblEntry::TraitVPtr(trait_ref) => write!(f, "TraitVPtr: {}", trait_ref)?,
```

Comment:
I would expect `Method({})` and `TraitVPtr({})`, so that they match the usual aspect.

Reply:
We have helpers for custom debug impls; you probably should use `f.debug_duple("Method").field(instance).finish()` etc

Reply:
Oh, the point is you want to use the `Display`, not `Debug` impls? That's a bit strange, maybe the `Debug` impls should instead of adjusted. But at least leave a comment why you need a manual impl.

Reply:
done.